### ISSUE TITLE
[BUG]: NullReferenceException thrown using IConnection.GetRaw() for JSON content in Octokit 8.0.1

### ISSUE THREAD

**Issue by martincostello** [octokit/octokit.net#2789]
### What happened?

Calling `IConnection.GetRaw()` for an endpoint that responds with non-binary content (e.g. `application/json`) results in a `NullReferenceException` being thrown when updating to Octokit 8.0.1.

This appears to be [due to the changes](https://github.com/octokit/octokit.net/pull/2782/files#diff-2787e44423613b17ad12829a18645c782b89a5d4fa3cd945274e103c7167e77aR704-R713) in #2782 to handle large binary responses.

In the case where the response is not a `Stream` but instead is just a `string`, `response.Body as Stream` returns `null`, and then causes an exception here:

https://github.com/octokit/octokit.net/blob/0238092ca0370f2edc2ee27e31b7c6e8ba747280/Octokit/Http/Connection.cs#L724

Fix incoming shortly, but a workaround is to use `IConnection.GetRawStream()` and refactor calling code to deal with `Stream` instead of `byte[]`.

### Versions

8.0.1

### Relevant log output

```shell
System.NullReferenceException: Object reference not set to an instance of an object.
    at Octokit.Connection.StreamToByteArray(Stream stream) in /_/Octokit/Http/Connection.cs:line 724
    at Octokit.Connection.GetRaw(IRequest request) in /_/Octokit/Http/Connection.cs:line 704
    at MartinCostello.Costellobot.Pages.DeliveryModel.OnGet(Int64 id) in /_/src/Costellobot/Pages/Webhooks/Delivery.cshtml.cs:line 64
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by martincostello** [comment]
Found through failing tests in these two Dependabot updates:

- https://github.com/martincostello/costellobot/pull/878
- https://github.com/martincostello/dependabot-helper/pull/891

**Comment by martincostello** [comment]
Also affects `IRepositoryContentsClient.GetRawContent()`, but in that case I can't workaround it myself.

**Comment by martincostello** [comment]
No problem! I did push up a fix in #2791 but I fat-fingered the number into the other fix for GitVersion and it got closed when you merged it 🤦🏻‍♂️
